File: app/src/main/java/com/worldventures/dreamtrips/wallet/service/lostcard/command/http/model/AddressRestResponse.java
package com.worldventures.dreamtrips.wallet.service.lostcard.command.http.model;

import android.support.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

import org.immutables.gson.Gson;
import org.immutables.value.Value;

import java.util.List;


@Value.Immutable
@Gson.TypeAdapters
public interface AddressRestResponse {

   @SerializedName("results")
   List<AddressComponents> results();
   @SerializedName("place_id")
   @Nullable
   String placeId();
   @SerializedName("status")
   String status();

   @Value.Immutable
   @Gson.TypeAdapters
   interface AddressComponents {
      @SerializedName("address_components")
      List<AddComponent> components();
      @SerializedName("formatted_address")
      String formattedAddress();
      @SerializedName("types")
      List<String> types();
   }

   @Value.Immutable
   @Gson.TypeAdapters
   interface AddComponent {
      @SerializedName("long_name")
      String longName();
      @SerializedName("short_name")
      String shortName();
      @SerializedName("types")
      List<String> types();
   }
}

File: wallet/src/main/java/com/worldventures/wallet/service/lostcard/command/http/model/AddressRestResponse.kt
package com.worldventures.wallet.service.lostcard.command.http.model

import com.google.gson.annotations.SerializedName

data class AddressRestResponse(
      @field:SerializedName("results") val results: List<ApiAddress>,
      @field:SerializedName("place_id") val placeId: String?,
      @field:SerializedName("status") val status: String
)

data class ApiAddress(
      @field:SerializedName("address_components") val components: List<AddressComponent>,
      @field:SerializedName("formatted_address") val formattedAddress: String,
      @field:SerializedName("types") val types: List<String>
)

data class AddressComponent(
      @field:SerializedName("long_name") val longName: String,
      @field:SerializedName("short_name") val shortName: String,
      @field:SerializedName("types") val types: List<String>)
